build_tools/cmake/iree_llvm.cmake

```diff
@@ -0,0 +1,201 @@
+#     does not require a source build of LLVM).
+
+macro(iree_llvm_configure_bundled)
+  message(STATUS "Adding bundled LLVM source dependency")
+  iree_llvm_set_bundled_cmake_options()
+
+  # Enable MLIR Python bindings if IREE Python bindings enabled.
+  if(IREE_BUILD_PYTHON_BINDINGS)
+    set(MLIR_ENABLE_BINDINGS_PYTHON ON)
+    set(MHLO_ENABLE_BINDINGS_PYTHON ON)
+  endif()
+
+  # Disable LLVM's warnings.
+  set(LLVM_ENABLE_WARNINGS OFF)
+
+  # Stash cmake build type in case LLVM messes with it.
+  set(_CMAKE_BUILD_TYPE "${CMAKE_BUILD_TYPE}")
+
+  add_subdirectory("third_party/llvm-project/llvm" "llvm-project" EXCLUDE_FROM_ALL)
+
+  # Reset CMAKE_BUILD_TYPE to its previous setting.
+  set(CMAKE_BUILD_TYPE "${_CMAKE_BUILD_TYPE}" )
+
+  # Set some CMake variables that mirror things exported in the find_package
+  # world. Source of truth for these is in an installed LLVMConfig.cmake,
+  # MLIRConfig.cmake, LLDConfig.cmake (etc) and in the various standalone
+  # build segments of each project's top-level CMakeLists.
+  set(LLVM_CMAKE_DIR "${IREE_BINARY_DIR}/llvm-project/lib/cmake/llvm")
+  list(APPEND CMAKE_MODULE_PATH "${LLVM_CMAKE_DIR}")
+  # TODO: Fix MLIR upstream so it doesn't spew into the containing project
+  # binary dir.
+  set(MLIR_CMAKE_DIR "${IREE_BINARY_DIR}/lib/cmake/mlir")
+  list(APPEND CMAKE_MODULE_PATH "${MLIR_CMAKE_DIR}")
+
+  set(LLVM_INCLUDE_DIRS
+    ${IREE_SOURCE_DIR}/third_party/llvm-project/llvm/include
+    ${IREE_BINARY_DIR}/llvm-project/include
+  )
+  set(MLIR_INCLUDE_DIRS
+    ${IREE_SOURCE_DIR}/third_party/llvm-project/mlir/include
+    ${IREE_BINARY_DIR}/llvm-project/tools/mlir/include
+  )
+  set(LLD_INCLUDE_DIRS
+    ${IREE_SOURCE_DIR}/third_party/llvm-project/lld/include
+    ${IREE_BINARY_DIR}/llvm-project/tools/lld/include
+  )
+
+  set(LLVM_BINARY_DIR "${IREE_BINARY_DIR}/llvm-project")
+  set(LLVM_TOOLS_BINARY_DIR "${LLVM_BINARY_DIR}/bin")
+  set(LLVM_EXTERNAL_LIT "${LLVM_TOOLS_BINARY_DIR}/llvm-lit")
+endmacro()
+
+macro(iree_llvm_configure_installed)
+  message(STATUS "Using installed LLVM components")
+  find_package(LLVM REQUIRED)
+  list(APPEND CMAKE_MODULE_PATH "${LLVM_CMAKE_DIR}")
+  include_directories(${LLVM_INCLUDE_DIRS})
+
+  find_package(MLIR REQUIRED)
+  list(APPEND CMAKE_MODULE_PATH "${MLIR_CMAKE_DIR}")
+  include_directories(${MLIR_INCLUDE_DIRS})
+
+  find_package(LLD REQUIRED)
+  list(APPEND CMAKE_MODULE_PATH "${LLD_CMAKE_DIR}")
+  include_directories(${LLD_INCLUDE_DIRS})
+
+  # Lit never gets installed with LLVM. So we have to reach into our copy
+  # of the monorepo to get it. I'm sorry. If this doesn't work for you,
+  # feel free to -DLLVM_EXTERNAL_LIT to provide your own.
+  # Note that LLVM style lit test helpers use LLVM_EXTERNAL_LIT, if provided,
+  # so this is consistent between the projects.
+  if(NOT LLVM_EXTERNAL_LIT)
+    set(LLVM_EXTERNAL_LIT "${IREE_SOURCE_DIR}/third_party/llvm-project/llvm/utils/lit/lit.py")
```

Comment:
So you still have to get the LLVM submodule even if you install it :-(

Reply:
Yep. There are other ways to come by lit though, and the user can manually specify LLVM_EXTERNAL_LIT to get around that requirement. Seems to be what the cool kids are doing.